.github/workflows/CI.yml

```diff
@@ -0,0 +1,40 @@
+name: CI
+on:
+  push:
+    branches: [master]
+    tags: [v*]
+  pull_request:
+
+jobs:
+  test:
+    name: Julia ${{ matrix.version }} - ${{ matrix.os }} - ${{ matrix.arch }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        version:
+          - '1.0'
+          - '1'
+        os:
+          - ubuntu-latest
+        arch:
+          - x64
```

Comment:
Since GH Actions can run a lot more jobs in parallel, I wonder whether it would make sense to test on more platforms. Especially 32-bit could be useful, because `Int`s not always being 64 can sometimes be a bit of a gotcha.

Reply:
Certainly, yeah, good idea. Do you think it's worth also testing on MacOS and Windows? Once I have everything working for ubuntu, I can add more to the grid.

Reply:
I don't think it would hurt. Since quite a bit of stuff depends on ChainRules nowadays, I think it would be nice to have broad test coverage. I could imagine that some IO functions we mark as `@nondifferentiable`, are defined a bit differently on Windows, so it wouldn't be impossible for there to be issues depending on the OS.